lib/cpp/src/thrift/server/TNonblockingServer.cpp

```diff
@@ -820,9 +821,11 @@ void TNonblockingServer::TConnection::setFlags(short eventFlags) {
  * Closes a connection
  */
 void TNonblockingServer::TConnection::close() {
-  // Delete the registered libevent
-  if (event_del(&event_) == -1) {
-    GlobalOutput.perror("TConnection::close() event_del", THRIFT_GET_SOCKET_ERROR);
+  if (eventFlags_) {
+    if (event_del(&event_) == -1) {
+      GlobalOutput("TConnection::close event_del");
```

Comment:
Sorry I copied this from another location and it must have been using GlobalOutput.  I will fix this.